flyteplugins/go/tasks/plugins/k8s/kfoperators/mpi/mpi.go

```diff
@@ -58,34 +55,33 @@ func (mpiOperatorResourceHandler) BuildResource(ctx context.Context, taskCtx plu
 		return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "nil task specification")
 	}
 
-	podSpec, objectMeta, primaryContainerName, err := flytek8s.ToK8sPodSpec(ctx, taskCtx)
-	if err != nil {
-		return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "Unable to create pod spec: [%v]", err.Error())
-	}
-	common.OverridePrimaryContainerName(podSpec, primaryContainerName, kubeflowv1.MPIJobDefaultContainerName)
-
-	var launcherReplica = common.ReplicaEntry{
-		ReplicaNum:    int32(1),
-		PodSpec:       podSpec.DeepCopy(),
-		RestartPolicy: commonOp.RestartPolicyNever,
-	}
-	var workerReplica = common.ReplicaEntry{
-		ReplicaNum:    int32(0),
-		PodSpec:       podSpec.DeepCopy(),
-		RestartPolicy: commonOp.RestartPolicyNever,
-	}
 	slots := int32(1)
 	runPolicy := commonOp.RunPolicy{}
 
+	var launcherReplicaSpec, workerReplicaSpec *commonOp.ReplicaSpec
+
 	if taskTemplate.TaskTypeVersion == 0 {
 		mpiTaskExtraArgs := plugins.DistributedMPITrainingTask{}
 		err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &mpiTaskExtraArgs)
 		if err != nil {
 			return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "invalid TaskSpecification [%v], Err: [%v]", taskTemplate.GetCustom(), err.Error())
 		}
 
-		workerReplica.ReplicaNum = mpiTaskExtraArgs.GetNumWorkers()
-		launcherReplica.ReplicaNum = mpiTaskExtraArgs.GetNumLauncherReplicas()
+		replicaSpec, err := common.ToReplicaSpec(ctx, taskCtx, kubeflowv1.MPIJobDefaultContainerName)
+		if err != nil {
+			return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "Unable to create replica spec: [%v]", err.Error())
+		}
+		launcherReplicaSpec = replicaSpec.DeepCopy()
+		// TODO (jeev): Is this even a valid configuration. Can there be more than 1
```

Comment:
Maintains parity for now, but will need to follow up in a subsequent investigation.